Update development mode  default caching docs (#52474)
rails/rails: guides/source/caching_with_rails.md

page and action caching you will need to add `actionpack-page_caching` and
`actionpack-action_caching` to your `Gemfile`.

> By default, action controller caching is only enabled in your production environment. You can play

Reviewer: We also have view caching. I _think_ that's action view?

Caching in Development
----------------------

> By default, caching is *enabled* in development mode, Rails uses
> [`:memory_store`](#activesupport-cache-memorystore) for caching.

Reviewer: ```suggestion
By default, caching is *enabled* in development mode with
[`:memory_store`](#activesupport-cache-memorystore).
```
